### PR TITLE
Automatic import of PES drivers

### PR DESCRIPTION
**Pull request by ceriottm**
Simplifies a bit the deployment of i-pi-driver PES files, as they get imported automatically without having to modify `__init.py__`. Also, added pes drivers for MACE and PET

### CODE REVIEW

**Comment by ceriottm** [comment]
Gawd, this was painful. The test infrastructure needs A LOT of love to be useful and usable. Hope the debugging here helps a bit.

**Comment by mahrossi** [comment]
> Gawd, this was painful. The test infrastructure needs A LOT of love to be useful and usable. Hope the debugging here helps a bit.

Well, it can become more useful, but it has been pretty useful already. 

The commit looks good to me and we will add on top of it as soon as it is merged.

File: drivers/py/pes/dummy.py [hunk 1]
```diff
@@ -1,4 +1,11 @@
+# this is how the driver will be referred to in the input files
+__DRIVER_NAME__ = "dummy"
+__DRIVER_CLASS__ = "Dummy_driver"
+
+
 class Dummy_driver(object):
+    """A dummy class providing the structure of an PES for the python driver."""
```
Comment (mahrossi) [inline drivers/py/pes/dummy.py:7]: an PES -> a PES

File: drivers/py/pes/mace.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+except:
+    MACECalculator = None
+
+__DRIVER_NAME__ = "mace"
+__DRIVER_CLASS__ = "MACE_driver"
+
+
+class MACE_driver(Dummy_driver):
+    def __init__(self, args=None):
+        self.error_msg = """Rascal driver requires specification of a .json model file fitted with librascal,
```
Comment (mahrossi) [inline drivers/py/pes/mace.py:19]: This error msg was not updated.

File: drivers/py/pes/mace.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                            and a template file that describes the chemical makeup of the structure.
+                            Example: python driver.py -m rascal -u -o model.json,template.xyz"""
+
+        super().__init__(args)
+
+    def check_arguments(self):
+        """Check the arguments requMACECalculatorred to run the driver
```
Comment (mahrossi) [inline drivers/py/pes/mace.py:26]: Something wrong in this line.

File: drivers/py/pes/mace.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        This loads the potential and atoms template in librascal
```
Comment (mahrossi) [inline drivers/py/pes/mace.py:28]: is it MACE or rascal?

File: drivers/py/pes/mace.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            arglist = self.args.split(",")
+        except ValueError:
+            sys.exit(self.error_msg)
+
+        self.model_atoms = read(arglist[0])
+        self.driver_example_atoms = arglist[0]
+        self.driver_model_path = arglist[1]
+
+    def __call__(self, cell, pos):
+        """Get energies, forces, and stresses from the librascal model"""
```
Comment (mahrossi) [inline drivers/py/pes/mace.py:40]: again, mace or rascal?

File: drivers/py/pes/mace.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        mace_calculator = MACECalculator(
+            model_path=self.driver_model_path, device="cpu"
+        )
+        atoms.set_calculator(mace_calculator)
+
+        pot = atoms.get_potential_energy()
+        force = atoms.get_forces()
+        stress = atoms.get_stress(voigt=False)
+
+        pot_ipi = float(unit_to_internal("energy", "electronvolt", pot))
```
Comment (mahrossi) [inline drivers/py/pes/mace.py:58]: is it always electronvolt? or should a comment be added that you expect MACE to have been trained on or return eV?

Comment (ceriottm) [reply drivers/py/pes/mace.py:58]: Didn't write this driver :-D

Comment (mahrossi) [reply drivers/py/pes/mace.py:58]: ok, but it was added in this commit. who do we need to tag?

**Comment by mahrossi** [comment]
Hello, it would be interesting to merge this too. Who is responsible for the MACE driver?

**Comment by ceriottm** [comment]
rebased after merging of #284/#294 . was painless

File: drivers/py/pes/mace.py (hunk 6 above)
Comment (mahrossi) [reply drivers/py/pes/mace.py:62]: I documented the units in the help string. It would be good to merge this, I think. Looks fine to me and works.

**Review by mahrossi — DISMISSED** [review]
ok!

**Comment by ceriottm** [comment]
I added a README, since @mahrossi  was already happy I'll merge this

**Pull request merged**